devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1357,36 +1455,56 @@ void HumanStateProvider::impl::computeSecondaryCalibrationRotationsForChain(cons
     for (auto const& jointZeroIdx: jointZeroIndices) {
         jointPos.setVal(jointZeroIdx, 0);
     }
+    // TODO check which value to give to the base (before we were using the base target measurement)
+    kinDynComputations->setRobotState(baseTransformSolution, jointPos, baseVel, jointVel, worldGravity);
+    
+    for (auto wearableTargetEntry : wearableTargets)
+    {
+        TargetName targetName = wearableTargetEntry.first;
+        ModelLinkName linkName = wearableTargetEntry.second->modelLinkName;
+        iDynTree::LinkIndex linkIndex = kinDynComputations->model().getLinkIndex(linkName);
 
-    kinDynComputations->setRobotState(linkTransformMatricesRaw.at(kinDynComputations->getFloatingBase()), jointPos, baseVel, jointVel, worldGravity);
+        std::cerr << "target: " << targetName << std::endl;
+        std::cerr << "link index" << linkIndex << std::endl;
```

Comment:
Is this a debug print? If it is meant to stay, probably it would be better using ``yarp`` logging.